app/src/main/java/com/spoony/spoony/core/network/ContentUriRequestBody.kt

```diff
@@ -0,0 +1,283 @@
+                }
+                return@use buffer.toByteArray()
+            }
+        }
+
+    private fun getOrientation(uri: Uri): Int {
+        contentResolver.query(
+            uri,
+            arrayOf(MediaStore.Images.Media.ORIENTATION),
+            null,
+            null,
+            null
+        )?.use { cursor ->
+            if (cursor.moveToFirst()) {
+                return cursor.getInt(cursor.getColumnIndexOrThrow(MediaStore.Images.Media.ORIENTATION))
+            }
+        }
+        return getExifOrientation(uri)
+    }
+
+    private fun getExifOrientation(uri: Uri): Int =
+        contentResolver.openInputStream(uri)?.use { input ->
+            ExifInterface(input).getAttributeInt(
+                ExifInterface.TAG_ORIENTATION,
+                ExifInterface.ORIENTATION_NORMAL
+            )
+        }?.let { orientation ->
+            when (orientation) {
+                ExifInterface.ORIENTATION_ROTATE_90 -> ORIENTATION_ROTATE_90
+                ExifInterface.ORIENTATION_ROTATE_180 -> ORIENTATION_ROTATE_180
+                ExifInterface.ORIENTATION_ROTATE_270 -> ORIENTATION_ROTATE_270
+                else -> ORIENTATION_NORMAL
+            }
+        } ?: ORIENTATION_NORMAL
+
+    private fun rotateBitmap(bitmap: Bitmap, angle: Int): Bitmap =
+        runCatching {
+            Matrix().apply {
+                postRotate(angle.toFloat())
+            }.let { matrix ->
+                Bitmap.createBitmap(
+                    bitmap,
+                    0,
+                    0,
+                    bitmap.width,
+                    bitmap.height,
+                    matrix,
+                    true
+                )
+            }
+        }.onSuccess { rotatedBitmap ->
+            if (rotatedBitmap != bitmap) {
+                bitmap.recycle()
+            }
+        }.getOrDefault(bitmap)
+
+    private fun calculateInSampleSize(width: Int, height: Int): Int {
+        val ratio = max(1, min(height / config.maxHeight, width / config.maxWidth))
+        return Integer.highestOneBit(ratio)
+    }
+
+    private fun calculateTargetSize(width: Int, height: Int): Size {
+        val ratio = width.toFloat() / height.toFloat()
+        return if (width > height) {
+            Size(config.maxWidth, (config.maxWidth / ratio).toInt())
+        } else {
+            Size((config.maxHeight * ratio).toInt(), config.maxHeight)
+        }
+    }
+
+    override fun contentLength(): Long = compressedImage?.size?.toLong() ?: -1L
+    override fun contentType(): MediaType? = metadata?.mimeType?.toMediaTypeOrNull()
+    override fun writeTo(sink: BufferedSink) {
+        compressedImage?.let(sink::write)
+    }
+
+    fun toFormData(name: String): MultipartBody.Part = MultipartBody.Part.createFormData(
+        name,
+        metadata?.fileName ?: DEFAULT_FILE_NAME,
+        this
+    )
+
+    companion object {
+        private const val ORIENTATION_NORMAL = 0
```

Comment:
p4) 이건 어떨지? Normal이라는게 의미가 모호한거같아여
```suggestion
        private const val ORIENTATION_DEFAULT = 0
```

Reply:
음~ 확실히 Default가 더 친숙하네요! 수정하겠습니당